### ISSUE TITLE
Metadata instance name not found

### ISSUE THREAD

**Issue by tchelovilar** [doitintl/janus#1]
Hi,

Thanks for the script, I tried to use it using the workload identity from the Kubernetes pod, but it doesn't work, because the metadata `instance/name` wasn't found. But I change it to `id` and works. 

https://github.com/doitintl/janus/blob/2577f2086a7cd04a23e3c94a1f4d86d2a58438a0/janus.py#L33

Can you check if with `id` also works from instance, because in that case you can change it to work in both ways.

Thanks

**Comment by Burekasim** [comment]
Hi @tchelovilar, 

Thanks for opening this issue, I will look into that.

**Comment by Burekasim** [comment]
Hi @tchelovilar 

Can you elaborate on your setup? is it GKE? 
I got the following output from the metadata server from my GKE cluster:
![image](https://user-images.githubusercontent.com/37156629/153221662-9be8eadb-5a1a-4396-923b-8c6b66d3ea00.png)

Can you share a screenshot from the node pool settings page? (please make sure to redact sensitive data), here is a screenshot from my playground: 
![image](https://user-images.githubusercontent.com/37156629/153222047-3963dbc8-790b-425f-9235-06a4b6b62732.png)

**Comment by Burekasim** [comment]
I just read your first message again, since you use workload identity, you use the gke metadata server, I will enable it and try again.

**Comment by tchelovilar** [comment]
Yes, I'm binding the kubernetes service account to the GCP service account:

https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity

**Comment by Burekasim** [comment]
I was able to reproduce the error, but I'm not sure that instance id is a good resource. 
I want that people will be able understand from cloudtrail logs which service/server initiate the api call. instance id is just a bunch of numbers. 

We can use hostname instead, it's available in GCP/GKE metadata server. the default structure contains the instance name, region and project name and trim the AWS role session name to 64 chars (that's AWS limit) to avoid errors from AWS API. 

What do you think @tchelovilar?

**Comment by Burekasim** [comment]
Hi @tchelovilar , 

I [pushed the modification](https://github.com/doitintl/janus/commit/7c454a62076b899b0d9bc2484fed1b2e51ce7a77), it works well for GCE, can you try it on GKE and update me?

**Comment by tchelovilar** [comment]
Hi @Burekasim ,

Works fine. Thank you.
